src/java/org/streamingpool/core/support/StreamSupport.java

```diff
@@ -41,8 +44,14 @@
 
     <T> OngoingLazyProviding<T> provide(StreamCreator<T> reactStream);
 
+    default Set<StreamId<?>> getSubgraphStartingFrom(StreamId<?> sourceId) {
```

Comment:
I think we never used the 'get' prefixes ... how about `upstreamGraphFrom` ...

Reply:
The returned streamIds are 'parents', or?

Reply:
could even call it `parentsUpstreamOf`.... Consider documenting this also: Is there any ordering guaranteed (I assume not, as it is a set)?, is the sourceId included in the returned set or not?

Reply:
leftover. Agree to use the same documentation of getAncestorsFrom in StreamDependencyTree since it is just delegation